digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/ProjectLinkDAO.scala

```diff
@@ -587,17 +589,23 @@ class ProjectLinkDAO {
     * @param userName: String - user name
     * @param discontinuity: Long - the discontinuity value to apply
     */
-  def updateProjectLinkNumbering(projectId: Long, roadNumber: Long, roadPart: Long, linkStatus: LinkStatus, newRoadNumber: Long, newRoadPart: Long, userName: String, discontinuity: Long): Unit = {
+  def updateProjectLinkNumbering(projectId: Long, roadNumber: Long, roadPart: Long, linkStatus: LinkStatus, newRoadNumber: Long, newRoadPart: Long, userName: String, discontinuity: Long, track: Track): Unit = {
     time(logger, "Update project link numbering") {
       val user = userName.replaceAll("[^A-Za-z0-9\\-]+", "")
       val sql = s"UPDATE PROJECT_LINK SET STATUS = ${linkStatus.value}, MODIFIED_BY='$user', ROAD_NUMBER = $newRoadNumber, ROAD_PART_NUMBER = $newRoadPart" +
         s"WHERE PROJECT_ID = $projectId  AND ROAD_NUMBER = $roadNumber AND ROAD_PART_NUMBER = $roadPart AND STATUS != ${LinkStatus.Terminated.value}"
       Q.updateNA(sql).execute
 
-      val updateLastLinkWithDiscontinuity =
+      val maxByTrack = fetchProjectLinks(projectId).filter(p => p.roadNumber == newRoadNumber && p.roadPartNumber == newRoadPart).groupBy(_.track).mapValues(p => p.maxBy(_.endAddrMValue))
+      val updateTrack = maxByTrack.filterNot(t => t._2.endAddrMValue == maxByTrack.minBy(_._2.endAddrMValue)._2.endAddrMValue && maxByTrack.minBy(_._2.endAddrMValue)._2.endAddrMValue != maxByTrack.maxBy(_._2.endAddrMValue)._2.endAddrMValue)
+      val updateLastLinkWithDiscontinuityString =  if(updateTrack.keys.contains(track)) {
```

Comment:
we can probably use collection checker here by for e.g. if(updateTrack.keys**.toList**.contains(track)) {